pyccel/codegen/printing/fcode.py

```diff
@@ -2219,8 +2227,17 @@ def _print_NumpyUfuncBase(self, expr):
 
     def _print_MathFunctionBase(self, expr):
         type_name = type(expr).__name__
+        if type_name == "MathPow":
+            code = self._print_Pow(expr)
+            return code
         func_name = math_function_to_fortran[type_name]
```

Comment:
Please add a try/except statement to catch possible `KeyError`s if the user requests a function which is not implemented. This can be used to raise a neat `PYCCEL_RESTRICTION_TODO`

Reply:
yes you are right. this will raise easy and readable error.